src/open_inwoner/accounts/tests/test_contact_views.py

```diff
@@ -85,10 +86,40 @@ def test_contact_filter_without_any_contacts(self):
             ),
         )
 
-    def test_contact_list_show_link_to_messages(self):
-        message_link = reverse("inbox:index", kwargs={"uuid": self.contact.uuid})
+    def test_messages_enabled_disabled(self):
+        """Assert that `Stuur Bericht` is displayed if and only if the message page is published"""
+
+        # case 1: no message page
         response = self.app.get(self.list_url, user=self.user)
-        self.assertContains(response, message_link)
+
+        self.assertNotIn("Stuur bericht", response)
+
+        # case 2: unpublished message page
+        page = api.create_page(
+            "Mijn Berichten",
+            "cms/fullwidth.html",
+            "nl",
+            slug="berichten",
+        )
+        page.application_namespace = "inbox"
+        page.save()
+
+        response = self.app.get(self.list_url, user=self.user)
+
+        self.assertNotIn("Stuur bericht", response)
+
+        # case 3: published message page
+        page.publish("nl")
+        page.save()
+
+        response = self.app.get(self.list_url, user=self.user)
+
+        html = response.content.decode("utf-8")
+        doc = PyQuery(html)
```

Comment:
FYI you can find-access the anchor tag by `response.pyquery("a[title='Stuur bericht']")`.

Reply:
I forgot that you can directly use pyquery on the response object. Nice!